### PR TITLE
21 custom device properties

### PR DESCRIPTION
**Pull request by ArtemKlots-in6k**

### CODE REVIEW

File: client/src/app/devices/device.service.ts [hunk 1]
```diff
@@ -15,6 +15,7 @@ export class DeviceService {
     }
 
     save(device: Device): Promise<Device> {
+        console.log(JSON.stringify(device));
```
Comment (ArtemKlots-in6k) [inline client/src/app/devices/device.service.ts]: Delete console.log

Comment (ArtemKlots-in6k) [inline client/src/app/devices/device.service.ts]: Done

File: client/src/app/devices/add-device.component.html [hunk 2]
```diff
@@ -41,10 +41,14 @@
...
         <div class="margin-block">
+            <custom-details [width]="441" [(device)]="device" ></custom-details>
+        </div>
+        <div class="margin-block">
             <md-input
                     class="notes"
                     placeholder="Примітки"
-                    [(ngModel)]="device.notes">
+                    [(ngModel)]="device.notes"
+                    >
```
Comment (alexanderko) [inline client/src/app/devices/add-device.component.html]: place in the same line with last tag

File: client/src/app/devices/details/custom-details.component.html [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<md-input
+        [style.width]="width"
+        class="custom-details"
+        placeholder="Додаткові параметри"
+        [(ngModel)]="customDetails.customDetails"
+>
```
Comment (alexanderko) [inline client/src/app/devices/details/custom-details.component.html]: place in the same line with last tag

File: client/src/app/devices/details/custom-details.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+export class CustomDetails{
+    public customDetails: string;
+
+    constructor (){
```
Comment (alexanderko) [inline client/src/app/devices/details/custom-details.ts]: missed space

File: client/src/app/devices/details/custom-details.component.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import {Component, Input, OnInit} from '@angular/core';
+import {Device} from '.././device';
+import {CustomDetails} from "./custom-details";
+
+@Component({
+    selector: 'custom-details',
+    templateUrl: 'app/devices/details/custom-details.component.html'
+})
+
+export class CustomDetailsBlock implements OnInit{
```
Comment (alexanderko) [inline client/src/app/devices/details/custom-details.component.ts]: missed space

File: client/src/app/devices/details/custom-details.component.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import {Component, Input, OnInit} from '@angular/core';
+import {Device} from '.././device';
+import {CustomDetails} from "./custom-details";
+
+@Component({
+    selector: 'custom-details',
+    templateUrl: 'app/devices/details/custom-details.component.html'
+})
+
```
Comment (alexanderko) [inline client/src/app/devices/details/custom-details.component.ts]: remove empty line

File: client/src/app/devices/add-device.component.html (hunk 2 above)
Comment (ArtemKlots-in6k) [inline client/src/app/devices/add-device.component.html]: Done

File: client/src/app/devices/details/custom-details.component.html (hunk 3 above)
Comment (ArtemKlots-in6k) [inline client/src/app/devices/details/custom-details.component.html]: Done

File: client/src/app/devices/details/custom-details.ts [hunk 7]
```diff
@@ -0,0 +1,7 @@
+export class CustomDetails{
```
Comment (alexanderko) [inline client/src/app/devices/details/custom-details.ts]: remove this class

File: client/src/app/devices/details/custom-details.component.ts [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+export class CustomDetailsBlock implements OnInit{
+    @Input('device') device: Device;
+    @Input('width') width: number;
+    customDetails: CustomDetails = new CustomDetails;
+
+    ngOnInit() {
```
Comment (alexanderko) [inline client/src/app/devices/details/custom-details.component.ts]: remove this init

File: client/src/app/devices/details/custom-details.component.ts (hunk 6 above)
Comment (ArtemKlots-in6k) [inline client/src/app/devices/details/custom-details.component.ts]: Done

File: client/src/app/devices/details/custom-details.ts (hunk 4 above)
Comment (YaroslavP-in6k) [inline client/src/app/devices/details/custom-details.ts]: done

File: client/src/app/devices/details/custom-details.ts (hunk 7 above)
Comment (YaroslavP-in6k) [inline client/src/app/devices/details/custom-details.ts]: resolve

File: client/src/app/devices/details/custom-details.component.ts (hunk 8 above)
Comment (YaroslavP-in6k) [inline client/src/app/devices/details/custom-details.component.ts]: resolve

**Pull request merged**